From f90768aeae4bd8f5d4a484bc1eb892201c7cbe26 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:54 -0400
Subject: [fs] nfsd: remove nfs4_file_put_fd

Message-id: <1410898849-22126-51-git-send-email-bfields@redhat.com>
Patchwork-id: 95776
O-Subject: [Patch RHEL7 050/165] nfsd: remove nfs4_file_put_fd
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

...and replace it with a simple swap call.

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 6d338b51eb6e37b4d6f1459c892f5ec7df0dad88
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index a19257f..c02bad6 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -394,15 +394,6 @@ nfs4_file_get_access(struct nfs4_file *fp, u32 access)
  return nfs_ok;
 }
 
-static struct file *nfs4_file_put_fd(struct nfs4_file *fp, int oflag)
-{
- struct file *filp;
-
- filp = fp->fi_fds[oflag];
- fp->fi_fds[oflag] = NULL;
- return filp;
-}
-
 static void __nfs4_file_put_access(struct nfs4_file *fp, int oflag)
 {
  might_lock(&fp->fi_lock);
@@ -411,9 +402,9 @@ static void __nfs4_file_put_access(struct nfs4_file *fp, int oflag)
   struct file *f1 = NULL;
   struct file *f2 = NULL;
 
-  f1 = nfs4_file_put_fd(fp, oflag);
+  swap(f1, fp->fi_fds[oflag]);
   if (atomic_read(&fp->fi_access[1 - oflag]) == 0)
-   f2 = nfs4_file_put_fd(fp, O_RDWR);
+   swap(f2, fp->fi_fds[O_RDWR]);
   spin_unlock(&fp->fi_lock);
   if (f1)
    fput(f1);
-- 
1.7.1